### PR TITLE
Retarget net462

### PR DESCRIPTION
**Pull request by chenriksson**
Depends on #6006 

Work in preparation for move to SQL AAD Authentication, including:
- Retargeting to "net462" (needed for token-based auth)
- Enabling AutoGenerateBindingRedirects + related cleanup
- Replacing packages.config with PackageRefs
- Dependency cleanup, including moving to same XUnit and Moq as other projects
- Moq-related test fixes

### CODE REVIEW

File: tests/NuGetGallery.Facts/Filters/ApiAuthorizeAttributeFacts.cs [hunk 1]
```diff
@@ -10,6 +10,8 @@
 using NuGetGallery.Authentication;
 using NuGetGallery.Security;
 using Xunit;
+using AuthorizationContext = System.Web.Mvc.AuthorizationContext;
+using AuthenticationTypes = NuGetGallery.Authentication.AuthenticationTypes;
```
Comment (chenriksson) [inline tests/NuGetGallery.Facts/Filters/ApiAuthorizeAttributeFacts.cs]: Resolves type conflicts that resulted from retargeting.

File: tests/NuGetGallery.Facts/Controllers/PackagesControllerFacts.cs [hunk 2]
```diff
@@ -163,15 +163,6 @@ public class PackagesControllerFacts
             return controller.Object;
         }
 
-        private static Mock<ISearchService> CreateSearchService()
-        {
-            var searchService = new Mock<ISearchService>();
-            searchService.Setup(s => s.Search(It.IsAny<SearchFilter>())).Returns(
-                (IQueryable<Package> p, string searchTerm) => Task.FromResult(new SearchResults(p.Count(), DateTime.UtcNow, p)));
```
[the remark this answers is not in the document]
Comment (dtivel) [reply tests/NuGetGallery.Facts/Controllers/PackagesControllerFacts.cs]: This is dead code and safe to remove, but should we add more `PackagesController` tests that verify `ISearchService` interaction?

File: src/NuGetGallery/Web.config [hunk 3]
```diff
@@ -135,107 +119,102 @@
...
-    <add key="Gallery.ExternalStatusUrl" value="https://status.nuget.org/" />
-    <add key="Gallery.ExternalAboutUrl" value="" />
-    <add key="Gallery.ExternalPrivacyPolicyUrl" value="" />
-    <add key="Gallery.ExternalTermsOfUseUrl" value="" />
-    <add key="Gallery.ExternalBrandingUrl" value="" />
-    <add key="Gallery.ExternalBrandingMessage" value="" />
-    <add key="Gallery.TrademarksUrl" value="" />
-    <add key="Gallery.EnforceDefaultSecurityPolicies" value="false" />
-
+    <add key="Gallery.ServiceDiscoveryUri" value="https://nugetdev0.blob.core.windows.net/v3-index/index.json"/>
```
Comment (loic-sharma) [inline src/NuGetGallery/Web.config]: Should `ServiceDiscoveryUri` and `SearchServiceResourceType` be left empty?

Comment (chenriksson) [reply src/NuGetGallery/Web.config]: Great catch... probably left this in when I was testing the search integration with the updates. Thanks!

**Comment by chenriksson** [comment]
@agr @loic-sharma @dtivel @skofman1 

Initial DEV deployments failed, but I fixed with commit 9017fdd80c6a490817de5f97567728998b196849. I had thought the binding redirects were not necessary due to `AutoGenerateBindingRedirects`, but they are still required for WAIISHost.exe (WA process that discovers the web role entry point).

I'm planning to redeploy DEV with SQL AAD PRs today, then will integrate these changes.

**Pull request merged**